Use single quotes to handle repository

ervansetiawan opened the pull request:
Repository that contains white spaces (e.g. a deb-s3 style repository such as http://repository trusty main) causes an issue during Packer execution.

This is an example of the error.
amazon-chroot: trusty: 1: trusty: Syntax error: Unterminated quoted string

duftler commented:
Is there something specific to the chroot builder that makes this change necessary?

I (and others) regularly use `debianRepository` values in rosco for aws (ebs) and gce bakes and haven't seen this issue.

I don't run the chroot builder myself, so I'm just looking for more insight as to why this issue would only affect this builder.

If it affects all the builders, we would need to make this change to all the templates (and would also like to know why we haven't run into this problem before).

ervansetiawan commented:
The problem only happens in the Packer chroot builder.

This is the bake request in Rush. Same request for both chroot and ebs builder.
`2016-04-05 13:26:37.601  INFO 4209 --- [ionThreadPool-4] c.n.s.r.l.scripts.ScriptExecutorLocal    : Executing 06379270-fb32-11e5-9e89-02e8b612ee99 with tokenized command: [sudo, packer, build, -color=false, -var, aws_region=us-west-1, -var, aws_ssh_username=ubuntu, -var, aws_instance_type=t2.micro, -var, aws_source_ami=ami-a64231c6, -var, aws_target_ami=siris-aws-all-20160405132636-ubuntu, -var, aws_associate_public_ip_address=true, -var, repository="http://repository trusty main", -var, package_type=deb, -var, packages=siris-aws, -var, configDir=/opt/rosco/config/packer, /opt/rosco/config/packer/aws-chroot.json]`

Packer log using aws-ebs.json (successful) : 
`2016/04/05 13:52:53 packer-builder-amazon-ebs: 2016/04/05 13:52:53 starting remote command: chmod +x /tmp/script_7943.sh; PACKER_BUILD_NAME='amazon-ebs' PACKER_BUILDER_TYPE='amazon-ebs' repository='http://repository trusty main' package_type='deb' packages='siris-aws' upgrade='' /tmp/script_7943.sh`

Packer log using aws-chroot.json (fails) :
`2016/04/05 13:54:55 packer-builder-amazon-chroot: 2016/04/05 13:54:55 Chroot execution exited with '2': 'chmod +x /tmp/script_7943.sh; PACKER_BUILD_NAME='amazon-chroot' PACKER_BUILDER_TYPE='amazon-chroot' package_type='deb' packages='siris-aws' upgrade='' repository=''"'"'http://repository trusty main'"'"'' disable_services=''"'"'true'"'"'' /tmp/script_7943.sh'`
trusty: 1: trusty: Syntax error: Unterminated quoted string

duftler commented:
Thanks for the details.

Looks reasonable to me. @gardleopard does this change look ok to you?

kalymero commented:
It makes sense, we found a similar issue internally. Wondering if it affect also the packages variable. @ervansetiawan did you try to pass more then one package?

duftler commented:
@ervansetiawan please rebase and add the same treatment for the `packages` parameter.

ervansetiawan commented:
@duftler done.

duftler commented:
Could you please rebase and squash commits instead of merging master?

ervansetiawan commented:
Okay, I will try it again. I rebased and squashed but it did not let me to push until I merge master. The merge master actually did not have any changes.

Obviously I did something wrong there ...

duftler commented:
Probably need -f when pushing since the branch was already pushed.
On Apr 5, 2016 7:40 PM, "Ervan Setiawan" <notifications@github.com> wrote:

> Okay, I will try it again. I rebased and squashed but it did not let me to
> push until I merge master. The merge master actually did not have any
> changes.
>
> Obviously I did something wrong there ...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/spinnaker/rosco/pull/83#issuecomment-206036908>
>

ervansetiawan commented:
Thanks. I think it is okay now. -f did it.

duftler commented:
LGTM.
Thanks for the contribution.
